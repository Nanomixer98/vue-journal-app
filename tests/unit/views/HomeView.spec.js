import { shallowMount } from '@vue/test-utils';
import Home from '@/views/HomeView';

describe('Pruebas en el Home View', () => {

    it('debe renderizar el componente correctamete', () => {
        let wrapper = shallowMount(Home)
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('hacer click en un botÃ³n debe redireccionar a no-entry', () => {

        const mockRouter = {
            push: jest.fn()
        }

        const wrapper = shallowMount(Home, {
            global: {
                mocks: {
                    $router: mockRouter
                }
            }
        })

        wrapper.find('button').trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({ name: 'no-entry' })

    })

})