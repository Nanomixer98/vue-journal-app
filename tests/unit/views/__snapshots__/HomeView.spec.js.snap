// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home View debe renderizar el componente correctamete 1`] = `
<div class="home"><img alt="Vue logo" src="">
  <h1>Hola Mundo</h1><button class="btn btn-primary">primary</button><button class="btn btn-secondary">secondary</button><button class="btn btn-success">success</button>
</div>
`;
